### ISSUE TITLE
Possible glibc issue on x86_64 linux systems

### ISSUE THREAD

**Issue by gotbadger** [Bearer/bearer#899]
## Description
Currently on linux it looks like we are seeing some segfaults with some users that have the `x86_64` instruction set. This is somewhat unexpected as we compile for `amd64` [here](https://github.com/Bearer/bearer/blob/main/.goreleaser/common.yaml#L52-L56) which should be compatible

## uname -a
works:
```
Linux hostname 5.10.0-21-amd64 #1 MP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux
Linux hostname 5.15.0-67-generic #74-Ubuntu SMP Wed Feb 22 14:14:39 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```
does not work:
```
Linux hostname 5.15.0-50-generic #56~20.04.1-Ubuntu SMP Tue Sep 27 15:51:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Linux hostname 5.13.0-1031-aws #35~20.04.1-Ubuntu SMP Mon Jun 13 22:30:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Linux hostname 5.15.0-69-generic #76~20.04.1-Ubuntu SMP Mon Mar 20 15:54:19 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```
## Related
https://stackoverflow.com/questions/75986529/bearer-scan-fails-to-run-for-react-project

**Comment by vjerci** [comment]
There a discussion here, explaining that for using statically linked binary you need the same version of glibc.
https://github.com/rust-lang/libc/issues/2054

**Comment by laluka** [comment]
Happy debugging folks, if this can help.. :)
https://gist.github.com/laluka/a9e36c46cd9be6884cfc34034d507e4a

**Comment by gotbadger** [comment]
We are trying to work out the best approach to this one as its a bit tricky. In the meantime if you have this issue you can always either:
- run the command via [docker run](https://docs.bearer.com/reference/installation/#docker)
- [compile](https://docs.bearer.com/contributing/code/) your own version for your system

**Comment by laluka** [comment]
Yup don't worry, I picked the docker right away, I was just giving feedbacks in case it'd help ;)

**Comment by gotbadger** [comment]
Currently working towards a solution here https://github.com/Bearer/bearer/pull/921

**Comment by gotbadger** [comment]
We've found a performance regression as a result of our fix and we are continuing to investigate and will release a new version as soon as we can. https://github.com/Bearer/bearer/pull/925

**Comment by didroe** [comment]
@laluka We've just released [1.4.0](https://github.com/Bearer/bearer/releases/tag/v1.4.0) which should fix this issue. Please can you try it out and let us know if it's resolved for you.

**Comment by laluka** [comment]
bearer_1.4.0_linux_amd64.tar.gz -> Works like a charm! ;)
